### ISSUE TITLE
Multi-GPU validation/test metrics count DistributedSampler padding duplicates

### ISSUE THREAD

**Issue by raghav-rathi** [roboflow/rf-detr#1507]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

Multi-GPU validation and test metrics count the images Lightning's `DistributedSampler` repeats to pad the split.

Lightning replaces the evaluation loader's `SequentialSampler` with a `DistributedSampler` that pads the index list to a multiple of `world_size` by repeating the leading indices (`indices += indices[:padding_size]`), so up to `world_size - 1` images are forwarded twice per epoch. `OnePassCocoMeanAveragePrecision.merge_distributed_state` (`src/rfdetr/training/coco_map.py:520-534` on `develop` @ 6b91a48e) concatenates every rank's state without dropping those repeats, and the macro-F1 accumulator and the keypoint OKS metric are fed the same way in `COCOEvalCallback.on_validation_batch_end` / `on_test_batch_end`. The legacy evaluator (`src/rfdetr/evaluation/coco_eval.py:444-457`) de-duplicated by `image_id` ("first rank to report an image_id owns it"); the Lightning path lost that.

Reproduced on 2x RTX 4090 with a 21-image `valid/` split, RF-DETR Nano, `devices=2, strategy="ddp"`, printing the merged state size after `merge_distributed_state`:

```
[rank 0] merge_distributed_state: local images=11 merged images=22
[rank 1] merge_distributed_state: local images=11 merged images=22
```

So `val/mAP*`, `val/mAR`, `val/F1` and their `test/` counterparts are computed over 22 images, one of them (image 0) twice. With fixed pretrained Nano weights at `eval_batch_size=1` (so batch composition cannot move the numbers), `RFDETR.evaluate(split="val")` on that split gives:

| | `val/mAP_50_95` | `val/mAP_50` | `val/mAR` |
| --- | --- | --- | --- |
| 1 GPU (21 images) | 0.0287 | 0.0644 | 0.1299 |
| 2 GPUs, `develop` (22 images) | 0.0286 | 0.0644 | 0.1286 |

The gap grows with `world_size` (an 8-GPU run like #1095 repeats up to 7 images) and shrinks with split size; it is a bias, not noise, and it also feeds `BestModelCallback` / early stopping decisions.

### Environment

- RF-DETR: 1.11.0.dev0 (`develop` @ 6b91a48e)
- OS: Ubuntu 22.04
- Python: 3.12
- PyTorch: 2.14.0+cu132, pytorch_lightning 2.6.6, NCCL
- GPU: 2x NVIDIA RTX 4090

### Minimal Reproducible Example

```python
# python repro.py  (Roboflow-layout dataset whose valid/ split has an odd number of images)
from rfdetr import RFDETRNano
from rfdetr.training import coco_map

_orig = coco_map.OnePassCocoMeanAveragePrecision.merge_distributed_state

def _merge(self):
    before = len(self.groundtruth_labels)
    _orig(self)
    print(f"merge_distributed_state: local images={before} merged images={len(self.groundtruth_labels)}", flush=True)

coco_map.OnePassCocoMeanAveragePrecision.merge_distributed_state = _merge

RFDETRNano().evaluate(split="val", dataset_dir="path/to/dataset", devices=2, strategy="ddp", eval_batch_size=1)
# merged images == len(valid) + 1 on a 2-GPU box
```

### Additional

Proposed fix: at each evaluation epoch start, read the loader's `DistributedSampler` (`rank`, `num_replicas`, `total_size`, `shuffle=False`) and skip accumulation for the positions at or past the dataset length; the forward still runs on the padded samples so every rank keeps an equal batch count and the epoch-end collectives stay symmetric (no switch to `UnrepeatedDistributedSampler`, which would make batch counts uneven). I have this implemented with unit tests against a real `DistributedSampler` and a gloo `ddp_spawn` end-to-end test (3 images on 2 ranks must merge to 3), and verified on the 2-GPU setup above that the two-GPU numbers now match the one-GPU numbers exactly; PR follows.

### Are you willing to submit a PR?

- [x] Yes, I'd like to help by submitting a PR!
